src/data/array_types.js

```diff
@@ -911,6 +914,8 @@ class PlacedSymbolStruct extends Struct {
     get crossTileID() { return this._structArray.uint32[this._pos4 + 10]; }
     set crossTileID(x: number) { this._structArray.uint32[this._pos4 + 10] = x; }
     get associatedIconIndex() { return this._structArray.int16[this._pos2 + 22]; }
+    get needsFlipping() { return this._structArray.uint8[this._pos1 + 46]; }
+    set needsFlipping(x: number) { this._structArray.uint8[this._pos1 + 46] = x; }
```

Comment:
Maybe rename `needsFlipping` to `flipState: undefined | flipRequired | flipNotRequired`? I think either is fine, but on my first time through the symbol code, I got a bit confused whether `needsFlip` indicated a *change* or whether it indicated a *state*.

Reply:
We will have to use the uint8 type as `{type: 'Uint8', name: 'flipState'}` in symbol_attributes.js, so that PlacedSymbolStruct could be regenerated based on this new descriptor.